kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconcilerTest.java

```diff
@@ -54,39 +57,116 @@ class KafkaProtocolFilterReconcilerTest {
+                .withUid("my-secret-uid")
+            .endMetadata()
+            .addToData("key", "value")
+            .build();
+
+    public static final Secret SECRET2 = new SecretBuilder()
+            .withNewMetadata()
+                .withName("my-secret2")
+                .withResourceVersion("abc")
+                .withUid("my-secret-uid2")
             .endMetadata()
             .addToData("key", "value")
             .build();
 
     public static final ConfigMap CONFIG_MAP = new ConfigMapBuilder()
             .withNewMetadata()
                 .withName("my-configmap")
-                .withGeneration(42L)
+                .withResourceVersion("def")
+                .withUid("my-configmap-uid")
+            .endMetadata()
+            .addToData("key", "value")
+            .build();
+
+    public static final ConfigMap CONFIG_MAP2 = new ConfigMapBuilder()
+            .withNewMetadata()
+                .withName("my-configmap2")
+            .withResourceVersion("hij")
+                .withUid("my-configmap-uid2")
             .endMetadata()
             .addToData("key", "value")
             .build();
     // @formatter:on
 
     public static List<Arguments> shouldSetResolvedRefs() {
         Context<KafkaProtocolFilter> bothExist = mock(Context.class);
-        when(bothExist.getSecondaryResourcesAsStream(Secret.class)).thenReturn(Stream.of(SECRET));
-        when(bothExist.getSecondaryResourcesAsStream(ConfigMap.class)).thenReturn(Stream.of(CONFIG_MAP));
+        when(bothExist.getSecondaryResources(Secret.class)).thenReturn(Set.of(SECRET));
+        when(bothExist.getSecondaryResources(ConfigMap.class)).thenReturn(Set.of(CONFIG_MAP));
 
         Context<KafkaProtocolFilter> secretExists = mock(Context.class);
-        when(secretExists.getSecondaryResourcesAsStream(Secret.class)).thenReturn(Stream.of(SECRET));
-        when(secretExists.getSecondaryResourcesAsStream(ConfigMap.class)).thenReturn(Stream.of());
+        when(secretExists.getSecondaryResources(Secret.class)).thenReturn(Set.of(SECRET));
+        when(secretExists.getSecondaryResources(ConfigMap.class)).thenReturn(Set.of());
 
         Context<KafkaProtocolFilter> cmExists = mock(Context.class);
-        when(cmExists.getSecondaryResourcesAsStream(Secret.class)).thenReturn(Stream.of());
-        when(cmExists.getSecondaryResourcesAsStream(ConfigMap.class)).thenReturn(Stream.of(CONFIG_MAP));
+        when(cmExists.getSecondaryResources(Secret.class)).thenReturn(Set.of());
+        when(cmExists.getSecondaryResources(ConfigMap.class)).thenReturn(Set.of(CONFIG_MAP));
 
         Context<KafkaProtocolFilter> neitherExists = mock(Context.class);
-        when(neitherExists.getSecondaryResourcesAsStream(Secret.class)).thenReturn(Stream.of());
-        when(neitherExists.getSecondaryResourcesAsStream(ConfigMap.class)).thenReturn(Stream.of());
+        when(neitherExists.getSecondaryResources(Secret.class)).thenReturn(Set.of());
```

Comment:
out of curiosity what prompted to the change of API call? `getSecondaryResourcesAsStream` -> `getSecondaryResources`?

Reply:
think it might have just been an intermediate step of the refactor before I introduced the Map, will go back to the stream